### PR TITLE
Fixed openid bug

### PR DESCRIPTION
**Pull request by CloudWise-Lukemiao**
Fix the exception of starting report when the configuration openid_url is HTTP and using keyclock to login without password abnormal bug

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java [hunk 1]
```diff
@@ -99,8 +99,7 @@ private static JSONObject getJWKFromProvider(String providerUrl)
     logger.debug("Using Provider Metadata: {}", providerMetadata);
 
     try {
-      URL url =
-          new URI(providerMetadata.getJWKSetURI().toString().replace("http", "https")).toURL();
+      URL url = new URI(providerMetadata.getJWKSetURI().toString().replace("http", "http")).toURL();
```
Comment (JulianFeinauer) [inline server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java:102]: Why do we generally need the replace here?

File: server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java [hunk 2]
```diff
@@ -189,7 +188,7 @@ private Claims validateToken(String token) {
         .setAllowedClockSkewSeconds(Long.MAX_VALUE / 1000)
         // .setSigningKey(DatatypeConverter.parseBase64Binary(secret))
         .setSigningKey(providerKey)
-        .parseClaimsJws(token)
+        .parseClaimsJws(token.replaceAll("access_token:", ""))
```
Comment (JulianFeinauer) [inline server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java:191]: What does this line do?

Comment (CloudWise-Lukemiao) [reply server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java:191]: I'm sorry, I take the return value( eg:{"access_token":xxx, "token_type": "Bearer","not-before-policy": 0, "session_state": "xxx","scope": "profile email","expires_in": 300,"refresh_expires_in": 1800,"refresh_token": ""} )as the token.

Comment (CloudWise-Lukemiao) [reply server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java:191]: I'm sorry, I take the return value( eg:{"access_token":xxx, "token_type": "Bearer","not-before-policy": 0, "session_state": "xxx","scope": "profile email","expires_in": 300,"refresh_expires_in": 1800,"refresh_token": ""} )as the token.

**Review by JulianFeinauer — APPROVED** [review]

**Comment by JulianFeinauer** [comment]
LGTM. Is the failing test systematic or should we care about it @jixuan1989 or @qiaojialin ? Otherwise this could be merged IMHO.

**Pull request merged**